docs/forms.rst

```diff
@@ -1,3 +1,93 @@
-
 Forms
 =====
+
+``is_core.forms`` contains helpers for forms rutines. The most interesting class is ``SmartForm``.
+
+SmartForm
+---------
+A base class for forms. Image that you want to call a method when a form is constructed. A basic approach is
+to override ``__init__`` method, which is pretty uncomftable because you have to call ``super`` and pass argument
+and etc. With ``SmartForm`` you can simply define a ``_init_fields`` method. But more interesting things are
+``_init_{field_name}`` method. The following code::
+
+    from is_core.forms import SmartForm
+
+    class MyForm(SmartForm):
+
+        def _init_fields():
+
+            for field in self.fields:
+                if field.name == 'foo':
+                    # do logic for foo
+                elif field.name == 'bar':
+                    # do logic for bar
+
+
+you can simply write in this way::
+
+    from is_core.forms import SmartForm
+
+    class MyForm(SmartForm):
+
+        def _init_foo(field):
+            # do logic for foo
+
+        def _init_bar(field):
+            # do logic for bar
+
+Yay! It's pretty, isn't it?
+
+
+``SmartForm`` contains helper methods that are called when an instance is created in a following order:
+
+1. ``_pre_init_fields``
+2. ``_init_{field_name}`` methods
+3. ``_init_fields``
+
+.. method:: SmartForm._pre_init_fields()
+
+This method is called when a form is created (after super ``__init__`` method and before ``_init_{field_name}`` methods).
+It make fields from ``base_required_fields`` attribute required.
+
+
+.. method:: SmartForm._init_{field_name}(field)
```

Comment:
Nejsem si jist, jaké jsou konvence pro psání Python dokumentace, ale neměl by tam být self, aby uživatel viděl, že ho může použít? Aby si někdo nemyslel, že je to statická funkce.